llvm/docs/DirectX/DXContainer.rst

```diff
@@ -400,3 +400,75 @@ SFI0 Part
 The SFI0 part encodes a 64-bit unsigned integer bitmask of the feature flags.
 This denotes which optional features the shader requires. The flag values are
 defined in `llvm/include/llvm/BinaryFormat/DXContainerConstants.def <https://github.com/llvm/llvm-project/blob/main/llvm/include/llvm/BinaryFormat/DXContainerConstants.def>`_.
+
+
+Root Signature (RST0) Part
+---------
+.. _RST0:
+
+The Root Signature defines the interface between the shader and the pipeline, specifying which resources are bound to the shader and how they are accessed. This structure serves as a contract between the application and the GPU, establishing a layout for resource binding that both the shader compiler and the runtime can understand.
+
+The Root Signature consists of a header followed by a collection of root parameters and static samplers. The structure uses a versioned design with offset-based references to allow for flexible serialization and deserialization.
+
+Root Signature Header
+~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: c
+
+   struct RootSignatureHeader {
+     uint32_t Version;
+     uint32_t NumParameters;
+     uint32_t ParametersOffset;
+     uint32_t NumStaticSamplers;
+     uint32_t StaticSamplerOffset;
+     uint32_t Flags;
+   }
+
+
+The `RootSignatureHeader` structure contains the top-level information about a root signature:
+
+- **Version**: Specifies the version of the root signature format. This allows for backward compatibility as the format evolves.
+- **NumParameters**: The number of root parameters contained in this root signature.
+- **ParametersOffset**: Byte offset from the beginning to the array of root parameters header.
```

Comment:
> Byte offset from the beginning to the array of root parameters header.

If all the offsets are from the beginning (I think they're all offset from the beginning of the RST0 part rather than the beginning of the container?) then maybe we can get away with just mentioning this once?